journals/apps/journals/blocks.py

```diff
@@ -83,6 +87,62 @@ class JournalRichTextBlock(blocks.RichTextBlock):
     def get_searchable_content(self, value):
         return [parser(value.source, 'html.parser').get_text(' ')]
 
+    def expand_db_html(self, html, for_editor=False, base_url='/'):
+        """
+        Override from wagtail.wagtailcore.rich_text to use full path
+        to embedded images
+        """
+        def replace_a_tag(m):
+            """
+            overridden, return href for Pages and Documents that
+            journals frontend knows how to display
+            """
+            attrs = extract_attrs(m.group(1))
+            if 'linktype' not in attrs:
+                # return unchanged
+                return m.group(0)
+            if attrs['linktype'] == 'page':
+                try:
+                    page = Page.objects.get(id=attrs['id'])
+                    return '<a href="{page_path}">'.format(page_path=page.specific.get_frontend_page_path())
+                except Page.DoesNotExist:
+                    return "<a>"
```

Comment:
Need to close tag.

Reply:
I copied this from wagtail handler. The closing tag gets put in after the link text by wagtail

Reply:
Gotcha. Didn't notice this. All good.